src/rpc/rpc-client.ts

```
@@ -0,0 +1,1205 @@
+ * });
+ * await client.connect();
+ */
+export class RpcClient implements RpcEventObservable {
+  public readonly networkId: NetworkId;
+
+  private client?: WebsocketHeartbeatJs;
+  private readonly resolver?: Resolver;
+  private readonly endpoint?: string;
+  private connectedPromise: BridgePromise<boolean>;
+  private requestPromiseMap: Map<number, BridgePromise<any>> = new Map();
+  private eventListeners: { event: keyof RpcEventMap | null; callback: Function }[] = [];
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)